src/openforms/submissions/tests/test_submission_report.py

```diff
@@ -98,6 +100,167 @@ def test_report_generation(self, mock_request):
         self.assertTrue(report.content.name.endswith("Test_Form.pdf"))
 
     def test_report_is_generated_in_same_language_as_submission(self):
+        # fixture_data
+        fields = [
+            # Component.type, user_input
+            ("bsn", "111222333"),
+            ("checkbox", "yes"),
+            ("currency", "1100000"),
+            ("date", "1911-06-29"),
+            ("email", "hostmaster@example.org"),
+            ("file", "download(2).pdf"),
+            ("iban", "NL56 INGB 0705 0051 00"),
+            ("licenseplate", "AA-00-13"),
+            ("map", "52.193459, 5.279538"),
+            ("number", "1"),
+            ("password", "Panda1911!"),  # XXX Why is this widget even an option?
+            ("phonenumber", "+49 1234 567 890"),
+            ("postcode", "3744 AA"),
+            ("radio", "Radio number one"),
+            ("select", "A fine selection"),
+            ("selectboxes", "This; That; The Other"),
+            ("signature", SIGNATURE),
+            ("textarea", "Largish predetermined ASCII"),
+            ("textfield", "Short predetermined ASCII"),
+            ("time", "23:50"),
+            ("updatenote", "C#"),
+        ]
+        submission_data = {}
+        translations = {}
+        components = []
+
+        # carthesian products are big, let's loop to fill these
+        for component_type, user_input in fields:
+            # common base
```

Comment:
perhaps defining some (pure python) factories in `openforms.formio.tests.factories` would help - you can use factory boy to generate/build [dicts](https://til.codeinthehole.com/posts/factory-boy-can-be-used-to-build-dicts/) :-)

That would then also line up better with the data structures used in `openforms.formio`